scheduler-worker/src/main/java/gov/nasa/jpl/aerie/scheduler/worker/SchedulerWorkerAppDriver.java

```diff
@@ -0,0 +1,121 @@
+    } catch (final IOException e) {
+      throw new Error("Failed to start SchedulingDSLCompilationService", e);
+    }
+
+    Runtime.getRuntime().addShutdownHook(new Thread(schedulingDSLCompilationService::close));
+
+    final var store = config.store();
+    if (!(store instanceof final PostgresStore postgresStore)) {
+      throw new UnexpectedSubtypeError(Store.class, store);
+    }
+    final var pgDataSource = new PGDataSource();
+    pgDataSource.setServerName(postgresStore.server());
+    pgDataSource.setPortNumber(postgresStore.port());
+    pgDataSource.setDatabaseName(postgresStore.database());
+    pgDataSource.setApplicationName("Scheduler Server");
+
+    final var hikariConfig = new HikariConfig();
+    hikariConfig.setUsername(postgresStore.user());
+    hikariConfig.setPassword(postgresStore.password());
+    hikariConfig.setDataSource(pgDataSource);
+
+    final var hikariDataSource = new HikariDataSource(hikariConfig);
+
+    final var stores = new Stores(
+      new PostgresSpecificationRepository(hikariDataSource),
+      new PostgresResultsCellRepository(hikariDataSource));
+
+    final var specificationService = new LocalSpecificationService(stores.specifications());
+    final var scheduleAgent = new SynchronousSchedulerAgent(specificationService,
+        merlinService,
+        merlinService,
+        config.merlinFileStore(),
+        config.missionRuleJarPath(),
+        config.outputMode(),
+        schedulingDSLCompilationService);
+
+    final var notificationQueue = new LinkedBlockingQueue<PostgresSchedulingRequestNotificationPayload>();
+    final var listenAction = new ListenSchedulerCapability(hikariDataSource, notificationQueue);
+    listenAction.registerListener();
+
+    final var app = Javalin.create().start(8080);
+    app.get("/health", ctx -> ctx.status(200));
+
+    while (true) {
+      final var notification = notificationQueue.take();
+      final var specificationRevision = notification.specificationRevision();
+      final var specificationId = new SpecificationId(notification.specificationId());
+
+      final Optional<ResultsProtocol.OwnerRole> owner = stores.results().claim(specificationId);
+      if (owner.isEmpty()) continue;
+
+      final var revisionData = new SpecificationRevisionData(specificationRevision);
+      final ResultsProtocol.WriterRole writer = owner.get();
+      try {
+        scheduleAgent.schedule(new ScheduleRequest(specificationId, revisionData), writer);
+      } catch (final Throwable ex) {
+        ex.printStackTrace(System.err);
+        writer.failWith(b -> b
+            .type("UNEXPECTED_SCHEDULER_EXCEPTION")
+            .message("Something went wrong while scheduling")
+            .trace(ex));
+      }
+    }
+  }
+
+  private static String getEnv(final String key, final String fallback){
+    final var env = System.getenv(key);
+    return env == null ? fallback : env;
+  }
+
+  private static WorkerAppConfiguration loadConfiguration() {
```

Comment:
Can you rename `WorkerAppConfiguration` in merlin and scheduler to `MerlinWorkerAppConfiguration` and `SchedulerWorkerAppConfiguration`? I know it technically doesn't matter, I just want it to be a little clearer

Reply:
While this is a minor change I think this would be best for a follow-up PR that makes this change in both `merlin-worker` and `scheduler-worker`.

We likely need to have a broader discussion about if we should / how much we should qualify subproject-specific names within subprojects that mirror other subprojects' architecture (for example `*-server`s and `*-worker`s having similar internal names).

Reply:
I'm a little weary about qualifying things to this degree though, mostly since I'd like naming conventions to be as consistent as possible.

For example, if we make this change then shouldn't we also change `PostgresResultsCellRepository` within both subprojects to `MerlinPostgresResultsCellRepository` and `SchedulerPostgresResultsCellRepository`? Maybe the answer is "yes", but it seems like a lot of clutter to add these qualifiers to every class that has the same name across multiple subprojects, ya know?

Reply:
Alright, I'm ok with it